libs/gl-cli/src/node.rs

```diff
@@ -0,0 +1,327 @@
+            label,
+            riskfactor,
+            maxfeepercent,
+            retry_for,
+            maxdelay,
+            exemptfee,
+            localinvreqid,
+            exclude,
+            maxfee,
+            description,
+        } => {
+            pay_handler(
+                config,
+                bolt11,
+                amount_msat,
+                label,
+                riskfactor,
+                maxfeepercent,
+                retry_for,
+                maxdelay,
+                exemptfee,
+                localinvreqid,
+                exclude,
+                maxfee,
+                description,
+            )
+            .await
+        }
+        Command::Stop => stop(config).await,
+    }
+}
+
+async fn log<P: AsRef<Path>>(config: Config<P>) -> Result<()> {
+    let creds_path = config.data_dir.as_ref().join(CREDENTIALS_FILE_NAME);
+    let creds = match util::read_credentials(&creds_path) {
+        Some(c) => c,
+        None => {
+            return Err(Error::CredentialsNotFoundError(format!(
+                "could not read from {}",
+                creds_path.display()
+            )))
+        }
+    };
+
+    let scheduler = gl_client::scheduler::Scheduler::new(config.network, creds)
+        .await
+        .map_err(Error::custom)?;
+
+    let mut node: gl_client::node::Client = scheduler.node().await.map_err(Error::custom)?;
+    let mut stream = node
+        .stream_log(StreamLogRequest {})
+        .await
+        .map_err(Error::custom)?
+        .into_inner();
+
+    //TODO:  Use tokio::select!() and ctrl + c to cancel listening for log lines
+    loop {
+        tokio::select! {
+            biased;
+            _ = tokio::signal::ctrl_c() => {
+                break;
+            }
+            maybe_line = stream.next() => {
+                match maybe_line {
+                    Some(line) => {
+                        println!("{}", line.map_err(Error::custom)?.line);
+                    },
+                    None => {
+                        break;
+                    },
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn stop<P: AsRef<Path>>(config: Config<P>) -> Result<()> {
+    let creds_path = config.data_dir.as_ref().join(CREDENTIALS_FILE_NAME);
+    let creds = match util::read_credentials(&creds_path) {
+        Some(c) => c,
+        None => {
+            return Err(Error::CredentialsNotFoundError(format!(
+                "could not read from {}",
+                creds_path.display()
+            )))
+        }
+    };
+
+    let scheduler = gl_client::scheduler::Scheduler::new(config.network, creds)
+        .await
+        .map_err(Error::custom)?;
+
+    let mut node: gl_client::node::ClnClient = scheduler.node().await.map_err(Error::custom)?;
+    let res = node
+        .stop(cln::StopRequest {})
+        .await
+        .map_err(|e| Error::custom(e.message()))?
+        .into_inner();
+    println!("{:?}", res);
```

Comment:
Notice that this will always produce an error, that we suppress in the python glcli as to not disconcert users. Both hiding and showing are options, but in this case the error is desired.